### ISSUE TITLE
Drop local preview feature completely

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#9729]
Local (offline) previews of posts and pages are currently useless, they're not showing what the post is going to look on the user's site. Local preview is just showing a post HTML output with a very simple CSS style and looks almost like the view in the visual/block editor.

Local previews are only available on local drafts. Local drafts will be very rare soon as we're going to synchronize them as soon as the app can, and local drafts will exist only while the app is in offline.

Note: Rendering the theme locally is way out of scope (it needs more than just a CSS file).

**Comment by maxme** [comment]
Update: we'll drop the "local preview" feature, but we can keep the "preview button", in this case the following scenario should happen while the app is offline:
- User edits an existing post, a new post or a draft,
- User taps “Preview”,
- App auto-saves the post locally,
- App notifies the user we can’t show a preview while offline.

**Comment by maxme** [comment]
Note: I added the "Previews" label on this one. When we finally close this ticket, we should browse all issues labeled "Previews" and close those related to local previews (most of them probably).

**Comment by yaelirub** [comment]
Epic ticket for removing local previews.

**Comment by diegoreymendez** [comment]
@maxme - I was trying to figure out if we have issues that request this work be done for pages, and the only one I found referencing pages specifically was this one.  Can you check if we need to create more specific issues for doing the same work for pages in Android?

**Comment by maxme** [comment]
Here his the list of tickets/tasks that needs to be done before we can close this one. In dependency order:

- [ ] Merge `master-remote-auto-save` in `develop` in FluxC.
- [ ] Hide remote Preview on Jetpack - https://github.com/wordpress-mobile/WordPress-Android/issues/10161
- [ ] Restore trashed posts - https://github.com/wordpress-mobile/WordPress-Android/issues/10106
- [ ] Implement remote post previews for pages like it's done for posts - https://github.com/wordpress-mobile/WordPress-Android/issues/9122
- [ ] Remove the local preview code `PostPreviewActivity` and `PostPreviewFragment`- https://github.com/wordpress-mobile/WordPress-Android/issues/10162
- [ ] Merge `feature/master-remote-previews` in `develop` in WPAndroid


Related tickets, but not necessary to close this issue:
- Make the Remote preview progress dialog cancellable - https://github.com/wordpress-mobile/WordPress-Android/issues/101410

**Comment by diegoreymendez** [comment]
While kind of relevant, I'm not sure how these two issues are a dependency to remove remote previews.

Restore trashed posts - #10106 
Restore trashed pages - #10219

The real dependency IMHO is that previews shouldn't be available in trashed posts which could be handled by showing a "Preview Unavailable" screen like in iOS.

We can then consider opening a separate issue to standardize across the iOS and Android UIs for trashed post lists (see below).

Do you agree @maxme ?

### iOS:

![IMG_0337](https://user-images.githubusercontent.com/1836005/61236076-83d99900-a70d-11e9-9732-8941d81bc449.PNG)

### Android:

![Screenshot_20190715-142839](https://user-images.githubusercontent.com/1836005/61236094-905df180-a70d-11e9-8651-302519f69ca1.png)

**Comment by maxme** [comment]
> The real dependency IMHO is that previews shouldn't be available in trashed posts which could be handled by showing a "Preview Unavailable" screen like in iOS.

Yes I agree we could solve it this way for previews. But it was also great to have the restore screen to avoid having trashed posts with local modifications (offline issue), so it's kind of fixing 2 issues at the same time.

The fix has already been merged for posts in #10185 - I'm not sure we should drop the ball now for pages.

**Comment by designsimply** [comment]
To make sure I understand correctly, the proposal here is to drop local previews while offline but previews while online would still continue to work the same way they do now?

**Comment by yaelirub** [comment]
@designsimply , By dropping local previews we mean dropping the "fake" previews and instead showing remote previews similarly to what we did in iOS:
https://github.com/wordpress-mobile/WordPress-iOS/issues/9282
